### PR TITLE
Fix issue with short-circuiting match all routes

### PR DESCRIPTION
**Pull request by howardjohn**
Fixes https://github.com/istio/istio/issues/26263 by reverting
https://github.com/istio/istio/pull/19471

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

**Comment by howardjohn** [comment]
@ramaraochavali your original PR had this comment: https://github.com/istio/istio/pull/19471#issuecomment-564088227

I am essentially reverting this behavior. Based on https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route_components.proto#envoy-v3-api-msg-config-route-v3-virtualhost route `The list of routes that will be matched, in order, for incoming requests. The first route that matches will be used.`. This means we *should* keep the non match all.

Please let me know if I misunderstand something here!

I also need to look into how the gateway merging works

File: pilot/pkg/networking/core/v1alpha3/route/route.go [hunk 1]
```diff
@@ -278,26 +278,23 @@ func BuildHTTPRoutesForVirtualService(
...
-					// We have a valid catch all route. No point building other routes, with match conditions.
-					out = append(out, r)
-					break
-				}
-			}
 			for _, match := range http.Match {
 				if r := translateRoute(push, node, http, match, listenPort, virtualService, serviceRegistry, gatewayNames); r != nil {
 					out = append(out, r)
+					// This is a catch all path. Routes are matched in order, so we will never go beyond this match
+					// As an optimization, we can just top sending any more routes here.
```
Comment (ramaraochavali) [inline pilot/pkg/networking/core/v1alpha3/route/route.go:294]: nit: /s/top/stop

**Review by ramaraochavali — APPROVED** [review]
Thanks. Makes sense

**Pull request merged**